editors/sc-ide/widgets/code_editor/editor.cpp

```diff
@@ -321,81 +328,93 @@ int GenericCodeEditor::findAll(const QRegExp& expr, QTextDocument::FindFlags opt
+    qsizetype i = 0;
+    while (i < str.size()) {
+        QRegularExpressionMatch backref_match = rexpr.match(str, i);
+        if (!backref_match.hasMatch())
+            break;
+        i = backref_match.capturedStart(0);
+        qsizetype len = backref_match.capturedLength(0);
+        if (backref_match.capturedStart(1) != -1) {
+            // qDebug("%lli (%s): escape", i, CSTR(backref_match.captured(1)));
             str.replace(i, len, "\\");
             i += 1;
-        } else if (rexpr.pos(2) != -1) {
-            QString num_str = rexpr.cap(2);
+        } else if (backref_match.capturedStart(2) != -1) {
+            QString num_str = backref_match.captured(2);
             num_str.remove(0, 1);
             int num = num_str.toInt();
-            // qDebug("%i (%s): backref = %i", i, CSTR(rexpr.cap(2)), num);
-            if (num <= expr.captureCount()) {
-                QString cap = expr.cap(num);
+            // qDebug("%lli (%s): backref = %i", i, CSTR(backref_match.captured(2)), num);
+            if (num <= match.lastCapturedIndex()) {
+                QString cap = match.captured(num);
                 // qDebug("resolving ref to: %s", CSTR(cap));
                 str.replace(i, len, cap);
                 i += cap.size();
             } else {
-                // qDebug("ref out of range", i, num);
+                // qDebug("%lli ref out of range: backref = %i", i, num);
                 str.remove(i, len);
             }
         } else {
-            // qDebug("%i (%s): unknown match", i, CSTR(rexpr.cap(0)));
+            // qDebug("%lli (%s): unknown match", i, CSTR(backref_match.captured(0)));
             str.remove(i, len);
         }
-        // qDebug(">> [%s] %i", CSTR(str), i);
+        // qDebug(">> [%s] %lli", CSTR(str), i);
     }
     // qDebug("END");
     return str;
 }
 
-bool GenericCodeEditor::replace(const QRegExp& expr, const QString& replacement, QTextDocument::FindFlags options) {
-    if (expr.isEmpty())
+bool GenericCodeEditor::replace(const QRegularExpression& expr, const QString& replacement,
+                                QTextDocument::FindFlags options) {
+    if (expr.pattern().isEmpty())
         return true;
 
     QTextCursor cursor = textCursor();
-    if (cursor.hasSelection() && expr.exactMatch(cursor.selectedText())) {
-        QString rstr = replacement;
-        if (expr.patternSyntax() != QRegExp::FixedString)
-            rstr = resolvedReplacement(rstr, expr);
-        cursor.insertText(rstr);
+    bool caps = (expr.patternOptions() & QRegularExpression::PatternOption::DontCaptureOption) == 0;
+    if (cursor.hasSelection()) {
+        QRegularExpressionMatch match = expr.match(cursor.selectedText());
+        if (match.hasMatch()) {
+            QString rstr = replacement;
+            if (caps) {
+                rstr = resolvedReplacement(rstr, match);
+            }
+            cursor.insertText(rstr);
+        }
```

Comment:
Does this block of code do more than what it is replacing? What does `caps` stand for? Is this doing something different with case sensitivity?

Reply:
Edit: just realised it stands for 'captures'. Perhaps write that out in full?

Reply:
It will perform regex backreference if `caps` is `true`. If `rstr` contains `\1`, `\2`, `\3` etc., `resolveReplacement()` function will return a new string with these backreferences(`\1`, `\2`...) replaced by their corresponding captured groups.

Reply:
Perhaps call it something like `replace_back_references_with_captures` ?